canonical-webteam/global-nav
# Modernise global-nav build infrastructure

## Done

- Renamed build folder to dist
- Added the dist folder to .gitignore, and ignored everything except the dist folder in .npmignore
- Replaced gulp build pipeline with rollup, which includes:
  - rollup-plugin-babel to transpile code
  - rollup-plugin-sass which processes the sass, and injects the global-nav css into `<head>` (so you only need to import the js file for functionality + styling)
- Set up eslint for the project, which just extends the airbnb rules
- Removed package-lock.json
- Updated lint script to use eslint and sass-lint.

## QA

- Pull code
- Run `yarn build` and check there are no errors
- Run `yarn watch` and open up `index.html` in a browser
- Check the global nav looks the same as before, and works properly
- Run `yarn lint` and check that it runs. There should be a million errors (they'll be fixed in later PRs)
- Run `npm pack` and check that the zip contains only the dist folder, package.json, readme and licence

## Issues

Fixes ubuntudesign/web-squad#747
